dhnx/simulation.py

```diff
@@ -10,23 +10,343 @@
         super().__init__(thermal_network)
         self.results = {}
 
-    def setup(self):
-        pass
+        self.nx_graph = self.thermal_network.to_nx_graph()
+
+        self.inc_mat = None
+
+        self.input_data = Dict()
+
+        self.rho = rho
+
+        self.c = c
+
+        self.mu = mu
+
+    def _concat_sequences(self, name):
+
+        select_sequences = [
+            d[name].copy().rename(columns=lambda x: component + '-' + x)
+            for component, d in self.thermal_network.sequences.items()
+            if name in d
+        ]
+
+        concat_sequences = pd.concat(select_sequences, 1)
+
+        return concat_sequences
+
+    def _prepare_hydraulic_eqn(self):
+
+        def _set_producers(m):
+            producers = [name for name in m.columns if re.search('producers', name)]
+
+            assert len(producers) == 1, "Currently, only one producer allowed."
+
+            m.loc[:, producers] = - m.loc[:, ~m.columns.isin(producers)].sum(1)
+
+            return m
+
+        self.inc_mat = nx.incidence_matrix(self.nx_graph, oriented=True).todense()
+
+        mass_flow = pd.DataFrame(
+            0,
+            columns=self.nx_graph.nodes(),
+            index=self.thermal_network.timeindex
+        )
+
+        input_data = self._concat_sequences('mass_flow')
+
+        mass_flow.loc[:, input_data.columns] = input_data
+
+        self.input_data.mass_flow = _set_producers(mass_flow)
+
+    def _solve_hydraulic_eqn(self, tolerance=1e-10):
+
+        edges_mass_flow = {}
+
+        for t in self.thermal_network.timeindex:
+
+            x, residuals, rank, s = np.linalg.lstsq(
+                self.inc_mat,
+                self.input_data.mass_flow.loc[t, :],
+                rcond=None
+            )
+
+            assert residuals < tolerance,\
+                f"Residuals {residuals} are larger than tolerance {tolerance}!"
+
+            edges_mass_flow.update({t: x})
+
+        self.results['edges-mass_flow'] = pd.DataFrame.from_dict(
+            edges_mass_flow,
+            orient='index',
+            columns=self.nx_graph.edges()
+        )
+
+        # TODO: Calculate these
+        # 'edges-pressure_losses'
+        # NOTE: edges have distributed and localized pressure losses zeta_dis * L/D**5
+        # nodes-pressure_losses
+        # NOTE: nodes have only localized pressure losses
+        # 'global-pressure_losses'
+        # 'producers-pump_power'
+
+        def _calculate_reynolds():
+            r"""
+            Calculates the reynolds number.
+
+            ..math::
+
+                Re = \frac{4\dot{m}{\pi\mu D}
+
+            Returns
+            -------
+
+            re : pd.DataFrame
+            """
+            edges_mass_flow = self.results['edges-mass_flow']
+
+            diameter = 1e-3 * self.thermal_network.components.edges['diameter_mm']
+
+            re = 4 * edges_mass_flow.divide(diameter, axis='index') \
+                / (np.pi * self.mu)
```

Comment:
I suspect that the divide function here is working differently than intended. If so, the mass flows are not divided by the correct diameter.

Here is how it is working right now with the example of the _producer_:

re[0] = 4 * mass_flow_prod[0] / diameter_prod / (pi * mu)
re[1] = 4 * mass_flow_prod[1] / diameter_cons_1 / (pi * mu)
re[2] = 4 * mass_flow_prod[2] / diameter_cons_2 / (pi * mu)

Reply:
Thanks for finding that. Fixed it with f6b8370.